app/controllers/concerns/health_patient_dashboard.rb

```diff
@@ -47,5 +47,75 @@ def determine_sort_order(medicaid_ids, column, direction)
       end
       raise 'Unknown sort column'
     end
+
+    def apply_filter
+      @search = search_setup(scope: :full_text_search)
+      @patients = @search.distinct if @search_string.present?
+
+      return unless params[:filter].present?
+
+      # Status Filter
+      @active_filter = true if params[:filter][:population] != 'all'
+      case params[:filter][:population]
+      when 'not_engaged'
+        @patients = @patients.not_engaged
+      when 'needs_f2f'
+        @patients = @patients.needs_f2f
+      when 'needs_qa'
+        @patients = @patients.needs_qa
+      when 'needs_intake'
+        @patients = @patients.needs_intake
+      when 'needs_renewal'
+        @patients = @patients.needs_renewal
+      end
+
+      # Needs filter
+      current_careplans = HealthPctp::Careplan.
+        where(patient_id: @patients.pluck(:id)).
+        order(created_at: :asc).
+        index_by(&:patient_id)
+
+      patients_with_needs = []
+      if params[:filter][:ncm_review].present?
+        needs_filter = true
+        patients_with_needs += current_careplans.select { |_, v| v.reviewed_by_rn_on.blank? }.values.map(&:patient_id)
+      end
+      if params[:filter][:manager_review].present?
+        needs_filter = true
+        patients_with_needs += current_careplans.select { |_, v| v.reviewed_by_ccm_on.blank? }.values.map(&:patient_id)
+      end
+      if params[:filter][:sent_to_pcp].present?
+        needs_filter = true
+        patients_with_needs += current_careplans.select { |_, v| v.sent_to_pcp_on.blank? }.values.map(&:patient_id)
+      end
+
+      if needs_filter
+        @active_filter = true
+        @patients = @patients.where(id: patients_with_needs)
+      end
+
+      # Team Member filter
+      if params[:filter][:user].present?
+        @active_filter = true
+        user_id = if params[:filter][:user] == 'unassigned'
+          nil
+        else
+          params[:filter][:user].to_i
+        end
+
+        @patients = @patients.where(care_coordinator_id: user_id)
+      end
+
+      if params[:filter][:nurse_care_manager_id].present? # rubocop:disable Style/GuardClause
+        @active_filter = true
+        nurse_care_manager_id = if params[:filter][:nurse_care_manager_id] == 'unassigned'
+          nil
+        else
+          params[:filter][:nurse_care_manager_id].to_i
+        end
+
+        @patients = @patients.where(nurse_care_manager_id: nurse_care_manager_id)
+      end
+    end
```

Comment:
Maybe return @patients?  I this looks like it should return a scope (or and maybe accept one) but basically is just a means to side-effects.